codegen/model/chart.go

```diff
@@ -0,0 +1,53 @@
+package model
+
+import v1 "k8s.io/api/core/v1"
+
+type Chart struct {
+	Operators []Operator
+
+	Values interface{}
+
+	// goes into the chart.yaml
+	Data Data
+}
+
+type Operator struct {
+	Name       string
+	Deployment Deployment
+	Args       []string
+}
+
+// values for Deployment template
+type Deployment struct {
+	Image     Image                    `json:"image,omitempty"`
+	Resources *v1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+type Image struct {
+	Tag        string        `json:"tag,omitempty"  desc:"tag for the container"`
+	Repository string        `json:"repository,omitempty"  desc:"image name (repository) for the container."`
+	Registry   string        `json:"registry,omitempty" desc:"image prefix/registry e.g. (quay.io/solo-io)"`
+	PullPolicy v1.PullPolicy `json:"pullPolicy,omitempty"  desc:"image pull policy for the container"`
+	PullSecret string        `json:"pullSecret,omitempty" desc:"image pull policy for the container "`
+
+	// options for building the image
+	Build *BuildOptions `json:"-"`
+}
+
+type BuildOptions struct {
+	// path to the main.go file
+	MainFile string
+
+	// push image after  build
+	Push bool
```

Comment:
Options like this are very often dependent on ENV variables, as we will want them to run in CI, but not locally. Wondering if we could use a library like viper, or envconfig, so that this value, and others like it can be configured manually, or via other sources.

Reply:
the idea is to be able to set the variable however you like, e.g.

```
operator := model.Operator{}

flag.BoolVar(&operator.Deployment.Image.Push, "push", false, "push the image for the operator")
```